modules.d/70overlayfs/prepare-overlayfs.sh

```diff
@@ -5,17 +5,94 @@ command -v getarg > /dev/null || . /lib/dracut-lib.sh
 getargbool 0 rd.overlayfs -d rd.live.overlay.overlayfs && overlayfs="yes"
 getargbool 0 rd.live.overlay.reset && reset_overlay="yes"
 
+overlayroot=$(getarg overlayroot=)
+
+root=$(getarg root=)
+if [ "${root%%:*}" = "live" ]; then
+    info "Live boot detected, skipping overlayfs setup (handled by dmsquash-live)"
+    return 0
+fi
+
+overlay_mode="tmpfs"
+overlay_device=""
+
+if [ -n "$overlayroot" ]; then
+    overlayfs="yes"
+
+    case "$overlayroot" in
+        tmpfs*)
+            overlay_mode="tmpfs"
+            ;;
+        /dev/* | LABEL=* | UUID=* | PARTLABEL=* | PARTUUID=*)
+            overlay_mode="device"
+            overlay_device=$(label_uuid_to_dev "$overlayroot")
```

Comment:
This function will echo the input (it doesn't return zero length) or return a path whether or not the partition exists.
You probably want to test `[ -b "$overlay_device ] || warn ...` before proceeding...

Same issue at line 39.